### PR TITLE
Rename Board.serial to Board.serial_number

### LINKED ISSUE #593: Rename serial to serial_number
Can possibly tie in with stuff from #368 

That would make serial numbers an interface, rather than an intrinsically required thing.

### PR DESCRIPTION
**Pull request by trickeydan**
## Checklist

- [x] I have updated the relevant documentation
- [x] I have added the `semver-major`, `semver-minor` or `semver-patch` label as appropriate
- [ ] I would like multiple reviewers to approve my code before merging

## Why do you want to make these changes?

- Makes the difference between a UART serial and serial number clear.
- Requested change from SR @j5api/brain-team 

## Which parts of the codebase do your changes affect?

Anything that uses a serial number

## How do your changes affect student or volunteer experience?

Makes the difference between a UART serial and serial number clear.

## Are your changes related to any existing issues or PRs? How so?

Fixes #593

### CODE REVIEW

File: j5/boards/sb/arduino.py [hunk 1]
```diff
@@ -24,6 +25,30 @@ def __init__(
...
+        pins = ImmutableDict[ArduinoUno.PinNumber, GPIOPin]({
+            **cast(Mapping[ArduinoUno.PinNumber, GPIOPin], self._analogue_pins),
+            **cast(Mapping[ArduinoUno.PinNumber, GPIOPin], self._digital_pins),
+
+        })
+        return pins
+
+    def make_safe(self) -> None:
+        """Make this board safe."""
+        pass
```
Comment (sersorrel) [inline j5/boards/sb/arduino.py]: this looks like an unrelated change?

Comment (trickeydan) [reply j5/boards/sb/arduino.py]: Literally force-pushed just now to get rid of that 😅

Comment (trickeydan) [reply j5/boards/sb/arduino.py]: (Was an artifact of a bodged rebase)
